lib/attrsets.nix

```diff
@@ -1798,6 +1799,51 @@ rec {
       then pkg.${output} or pkg.out or pkg
       else pkg;
 
+  /**
+    Get the first of the `outputs` provided by the package, or the default.
+    This function is alligned with `_overrideFirst()` from the `multiple-outputs.sh` setup hook.
+    Like `getOutput`, the function is idempotent.
+
+    # Inputs
+
+    `outputs`
+
+    : 1\. Function argument
+
+    `pkg`
+
+    : 2\. Function argument
+
+    # Type
+
+    ```
+    getFirstOutput :: [String] -> (Derivation | DerivationOutput) -> DerivationOutput
```

Comment:
CC @emilazy I ended up changing all of the signatures to what I think they really are. The name `DerivationOutput` is somewhat arbitrary, maybe I should document that `DerivationOutput` is `Derivation` augmented with `{ outputName: String, outputSpecified: Bool }`, but I doubt this is the right place to communicate this

Reply:
I think we could just say `Derivation` here.

Reply:
I'd be fine with both `Derivation -> DerivationOutput` and `Derivation -> Derivation`. They'd be both technically correct, increasingly more ambiguous, increasingly easier to read

Reply:
I went for the simpler and less precise one